vspec/__init__.py

```diff
@@ -247,13 +252,34 @@ def cleanup_deep_model(deep_model):
     if "$name$" in deep_model:
         del deep_model['$name$']
 
-    if (deep_model["type"] == "branch"):
+    # children as of today exists only for branches
+    if "children" in deep_model:
         children = deep_model["children"]
         for child in deep_model["children"]:
             cleanup_deep_model(children[child])
 
     return None
 
+#
+# Meta data on extended attributes needs to be cleaned as part of the second cleaning step.
+# as it is not included in first step.
+#
+def clean_metadata(node):
+
+
+    def clean_dict(node):
+
+        for k in list(node.keys()):
+            if k in ["$file_name$", "$line$"]:
+                del node[k]
+            elif isinstance(node[k],dict):
+                clean_dict(node[k]);
+
+    if isinstance(node,VSSNode):
+        clean_dict(node.extended_attributes)
```

Comment:
This is an alternative approach, cleaning extended attributes when all handling are finished. That we could potentially do also for other metadata but then more refactoring would be needed. Today we get warnings on e.g. `$line$` (not a defined attribute) if kept and removed first here.